### PR TITLE
Add webkit recording utility and implement for exercises.

### PR DESCRIPTION
**Pull request by eschlange**
* Stripped existing Flash recording utilities.
* Added new HTML5 recording utilities.
* Implemented the recording process in audio and multisyllabic views.

[#102222452]

### CODE REVIEW

**Comment by eschlange** [comment]
@michaelwehrley ```recorderHTML5.js``` and ```recorderWorker.js``` are imported Javascript files from the "https://github.com/mattdiamond/Recorderjs" project.

**Comment by ericcf** [comment]
@eschlange @michaelwehrley FYI the convention for 3rd party JS is to put it in the `vendor/assets/javascripts` folder. You can include it in the manifest file (application.js).

**Comment by eschlange** [comment]
@ericcf Thanks, Eric! I'll move those files now.

File: app/assets/javascripts/audio-exercise-utilities.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+function initRecorder() {
+  try {
+    window.AudioContext = window.AudioContext || window.webkitAudioContext;
+    navigator.getUserMedia = (navigator.getUserMedia ||
+                              navigator.webkitGetUserMedia ||
+                              navigator.mozGetUserMedia ||
+                              navigator.msGetUserMedia);
+    window.URL = window.URL || window.webkitURL;
+    audio_context = new AudioContext;
```
Comment (michaelwehrley) [inline app/assets/javascripts/audio-exercise-utilities.js]: should this be `new AudioContext();`?

File: app/assets/javascripts/recorderHTML5.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
+(function(window){
```
Comment (michaelwehrley) [inline app/assets/javascripts/recorderHTML5.js]: Is this a library? And was @ericcf was referring to?

File: app/views/multisyllabic_decks/_paginated_deck.html.erb [hunk 3]
```diff
@@ -1,55 +1,73 @@
...
+  </div>
+</div>
+
+<script>
+  var audio_context;
+  var recorder;
+  var recordingHTML5 = false;
+  var timeStart = new Date();
+
+  function toggleRecordingHTML5(button) {
```
Comment (michaelwehrley) [inline app/views/multisyllabic_decks/_paginated_deck.html.erb]: did you think about putting all this js in a js folder and setting it to `app` like you did above?

File: app/assets/javascripts/audio-exercise-utilities.js (hunk 1 above)
Comment (eschlange) [inline app/assets/javascripts/audio-exercise-utilities.js]: This segment was borrowed from the example given in the recorder.js project: https://github.com/mattdiamond/Recorderjs/blob/master/example_simple_exportwav.html#L92

The example and the Rogalski exercises are functional with ```new AudioContext;```

File: app/views/multisyllabic_decks/_paginated_deck.html.erb (hunk 3 above)
Comment (michaelwehrley) [inline app/views/multisyllabic_decks/_paginated_deck.html.erb]: This is probably a personal preference, but I like to keep as much js - especially defining js functions - out of html.

File: app/views/audio_decks/_paginated_deck.html.erb [hunk 4]
```diff
@@ -16,39 +20,70 @@
...
+  var recordingHTML5 = false;
+  var timeStart = new Date();
+  var hintCount = 0;
+
+  function hideHintButton() {
+    $('#hint-link').addClass("hidden");
+  }
+
+  function revealHint() {
+    var MAX_HINT_COUNT = 3;
```
Comment (michaelwehrley) [inline app/views/audio_decks/_paginated_deck.html.erb]: I don't think this matters here, but in ruby, I believe it better practice to set the constant outside the function.  I don't think there is a reason to redefine the constant every time the function gets called.

File: app/views/audio_decks/_paginated_deck.html.erb [hunk 5]
```diff
@@ -16,39 +20,70 @@
...
+      button.innerHTML = "<i class='fa fa-circle large-fa-icon red-fa-icon'></i><br>" +
+      "Recording in Progress<br>Press when your description is complete.";
+    }
+    recordingHTML5 = !recordingHTML5;
+  }
+
+  window.WORKER_PATH = "<%= asset_path('recorderWorker.js') %>";
+  initRecorder();
+</script>
+
```
Comment (michaelwehrley) [inline app/views/audio_decks/_paginated_deck.html.erb]: note an extra line.

File: app/views/multisyllabic_decks/_paginated_deck.html.erb [hunk 6]
```diff
@@ -1,55 +1,73 @@
...
-		<div class="pagination-link hidden">
-			<a id="next-button" href="#"> Next Word <i class='fa fa-chevron-right'></i></a>
-		</div>
-		<div class="content" style="text-align:center;">
-			<h1><%= @multisyllabic_card.word %></h1>
+  <div class='col_4 carton centered'>
+    <div class="pagination-link hidden">
+      <a id="next-button" href="#"> Next Word <i class='fa fa-chevron-right'></i></a>
+    </div>
+    <div class="content" style="text-align:center;">
```
Comment (michaelwehrley) [inline app/views/multisyllabic_decks/_paginated_deck.html.erb]: if you are using bootstrap, there is class `text-center`

**Comment by michaelwehrley** [comment]
Besides - I imagine - are a few stylistic notes; this seems :solid to me.  @eschlange:merge when you desire.

**Comment by eschlange** [comment]
@michaelwehrley Thanks for the review, Mike! All of your suggestions looks good, I'll make updates and merge after they are complete.

**Pull request merged**